content/posts/react-query-and-type-script/index.mdx

```diff
@@ -344,6 +344,23 @@ function useGroup(id: number | undefined) {
 }
 ```
 
+<Aside title="Update" icon="bell">
+
+Since v5.25, You can use [`skipToken`](https://tanstack.com/query/latest/docs/framework/react/guides/disabling-queries#typesafe-disabling-of-queries-using-skiptoken) to disable a query while having good type safety in `queryFn`:
+
+```ts:title=skipToken
+import { useQuery, skipToken } from "@tanstack/query";
+
+function useGroup(id: number | undefined) {
+  return useQuery({
+    queryKey: ['group', id],
+    queryFn: id ? () => fetchGroup(id) : skipToken,
+    enabled: Boolean(id),
```

Comment:
```suggestion
```

Reply:
Will this write to the cache? I mean for `['group', undefined]`

Reply:
yes, just like `enabled: false` does. We actually set enabled internally.